### PR TITLE
Transfer Issues to Other Repositories

### PR DESCRIPTION
**Pull request by srafi1**
Implements transferring issues to other repositories as requested in issue #2252. This feature differs from the feature request in one way: the user only specifies TARGET_REPO rather than USER/TARGET_REPO. This is because users cannot move issues between different users or organizations according to [this guide](https://help.github.com/en/github/managing-your-work-on-github/transferring-an-issue-to-another-repository).

Format:
```
hub issue transfer <ISSUE_NUMBER> <TARGET_REPO>
```

Example:
```
hub issue transfer 2252 hub-fork
```

### CODE REVIEW

File: commands/issue.go [hunk 1]
```diff
@@ -710,3 +720,105 @@ func milestoneValueToNumber(value string, client *github.Client, project *github
...
+                }
+            }
+        }`, issueNumber, owner, currRepo)
+	body := make(map[string]interface{})
+	body["query"] = query
+
+	response, err := gh.GenericAPIRequest("POST", "graphql", body, nil, 0)
+	utils.Check(err)
+
+	responseData := make(map[string]interface{})
```
Comment (mislav) [inline commands/issue.go]: You can have this be:

```go
responseData := struct {
  Data struct {
    Repository struct {
      Issue struct {
        ID string `json:"id"`
      }
    }
  }
  Errors []struct {
    Message string
  }
}{}
```

And then you can umarshal into that and read `responseData.Data.Repository.Issue.ID` directly without using type assertions.

File: commands/issue.go [hunk 2]
```diff
@@ -710,3 +720,105 @@ func milestoneValueToNumber(value string, client *github.Client, project *github
...
+	response, err = gh.GenericAPIRequest("POST", "graphql", body, nil, 0)
+	utils.Check(err)
+
+	responseData = make(map[string]interface{})
+	err = response.Unmarshal(&responseData)
+	utils.Check(err)
+
+	if _, keyExists := responseData["errors"]; keyExists {
+		ui.Errorf("Error transferring the issue\n")
+		os.Exit(1)
```
Comment (mislav) [inline commands/issue.go]: Tip: `utils.Check(err)` does the printing and `os.Exit(1)` combo for you 😉

File: commands/issue.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ issue [-a <ASSIGNEE>] [-c <CREATOR>] [-@ <USER>] [-s <STATE>] [-f <FORMAT>] [-M
 issue show [-f <FORMAT>] <NUMBER>
 issue create [-oc] [-m <MESSAGE>|-F <FILE>] [--edit] [-a <USERS>] [-M <MILESTONE>] [-l <LABELS>]
 issue labels [--color]
+issue transfer <ISSUE_NUMBER> <TARGET_REPO>
```
Comment (mislav) [inline commands/issue.go]: Nit: we prefer using hyphens here 
```suggestion
issue transfer <ISSUE-NUMBER> <TARGET-REPO>
```

**Comment by mislav** [comment]
This also needs tests in order to have CI passing again. You can find existing tests in `features/issue.feature` and run them with `bin/cucumber features/issue.feature`, but we don't have any examples of tests that handle GraphQL API calls since we didn't use GraphQL before this. If this proves to hard for you to write, just holler and I will take over. 🙇

**Comment by srafi1** [comment]
Thanks for the feedback! I implemented the method for unmarshalling that you suggested and the code looks much cleaner now without type assertions. I haven't tried writing any tests yet, so I will try that (hopefully during the weekend).

File: commands/issue.go [hunk 4]
```diff
@@ -710,3 +721,106 @@ func milestoneValueToNumber(value string, client *github.Client, project *github
...
+	project, err := localRepo.MainProject()
+	utils.Check(err)
+
+	owner := project.Owner
+	currRepo := project.Name
+	host := project.Host
+	gh := github.NewClient(host)
+
+	query := fmt.Sprintf(`
+        query($issue: Int = %s, $owner: String = "%s", $repo: String = "%s") {
```
Comment (mislav) [inline commands/issue.go]: Even though this works, the query should ideally not contain any values:
```suggestion
        query($issue: Int!, $owner: String!, $repo: String!) {
```
instead, pass values as `variables` parameter in JSON:
```go
variables := map[string]interface{}{
  "issue": issueNumber,
  "owner": owner,
  "repo": currRepo,
}
body["variables"] = variables
```

**Comment by srafi1** [comment]
I put variables in their own parameter, but I haven't written any tests. I looked at `steps.rb` and `issue.feature`, but I'm not familiar with Ruby or Cucumber, so I don't know how to write them. Can you help me out with this?

**Comment by mislav** [comment]
@srafi1 Great work; thank you! I've also added some cleanups and tests.

One thing that only ocurred to me after I took over your branch was that the first 2 GraphQL queries can be performed in a single request (rather than over 2 requests), so I've implemented that.

I've also added a specialized `Client.GraphQL` function to move us away from `GenericAPIRequest` since the latter was only intended for the `hub api` command.

**Pull request merged**